core/types/src/svn.rs

```diff
@@ -27,6 +28,30 @@ impl_newtype! {
 #[derive(Default, Debug, Clone, Hash, PartialEq, Eq)]
 pub struct CpuSvn(sgx_cpu_svn_t);
 
-impl_newtype_for_bytestruct! {
+impl Display for CpuSvn {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(f, "CpuSvn: {:X}", self)
```

Comment:
:question: would it be better to update the display in `impl_newtype_for_bytestruct` to be of the form:
```rust
        impl ::core::fmt::Display for $wrapper {
            fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
                write!(f, "{}: ", stringify!($wrapper))?;
                ::core::fmt::UpperHex::fmt(self, f)
            }
        }
```
Note: used multiline write/fmt as this allows the passing of the alternate flag unconditionally via the `Formatter`, https://doc.rust-lang.org/std/fmt/trait.UpperHex.html

Reply:
Display should not include the name of the struct.

Reply:
Also, AFAIK, `UpperHex` will print `[0xDE, 0xAD, 0xBE, 0xEF]`, which isn't what we actually want, FWIW.

Reply:
What's the impetus to move display out of the macro if it's going to be implemented the same as the macro was?
I can understand it for the other types where you want to show flags instead of a `u32` or perhaps a byte array instead of an integer.
Is it so that *all* the Display impls are explicit?

Reply:
When I started on this stack of PRs, I wasn't sure if a blanket macro implementation would make sense because I thought each `Display` might be slightly different. Now that all of them are done, I agree that they can all use the same macro impl.